ext/pdo/pdo_dbh.c

```diff
@@ -38,6 +38,51 @@
 
 static int pdo_dbh_attribute_set(pdo_dbh_t *dbh, long attr, zval *value TSRMLS_DC);
 
+PDO_API zval *pdo_throw_exception_ex(zend_class_entry *exception_ce, pdo_error_type *pdo_err, long code TSRMLS_DC, const char *format, ...) /* {{{ */
+{
+	va_list arg;
+	char *message;
+	zval *zexception, *info;
+	zend_class_entry *default_exception_ce;
+
+	va_start(arg, format);
+	zend_vspprintf(&message, 0, format, arg);
+	va_end(arg);
+
+	default_exception_ce = zend_exception_get_default(TSRMLS_C);
+
+	MAKE_STD_ZVAL(zexception);
+	if (exception_ce) {
```

Comment:
Why these checks are necessary if it's an internal function? It also always gets php_pdo_get_exception() anyway.